modules/sw360/sw360-workflow/src/main/java/org/eclipse/sw360/antenna/sw360/SW360MetaDataUpdater.java

```diff
@@ -108,15 +112,74 @@ public boolean isUploadSources() {
     public SW360Release uploadAttachments(SW360Release sw360Release, Map<Path, SW360AttachmentType> attachments) {
         AttachmentUploadRequest.Builder<SW360Release> builder = AttachmentUploadRequest.builder(sw360Release);
         for (Map.Entry<Path, SW360AttachmentType> e : attachments.entrySet()) {
-            builder = builder.addAttachment(e.getKey(), e.getValue());
+            if (needUpload(sw360Release, e.getKey(), e.getValue())) {
```

Comment:
I am just wondering if this check wouldn't be better placed in the sw360 client, maybe the attachmentuploadrequest. Since this is something that should be applied to every kind of attachment uploaded, no matter which halresource it goes with and it is not an antenna specific check, but a sw360 specific one.

Reply:
Not sure about this. Actually, the additional check is just an optimization, which we can apply because we know our workflows. It is not clear whether this is possible in general.
For instance, the SW360Release entity that is target of the upload may not be up-to-date because of other updates on the server; then the check would be done on stale data. To be on the safe side, the most recent entity would have to be queried from the server first; but then it is easier to do the upload directly and let SW360 fail based on its logic (which after all is the ultimo ratio). - It would be nice, however, if SW360 returned a meaningful error message rather than throwing a generic 500 error when it does not like a specific attachment upload.

Reply:
Fair enough.